### PR TITLE
test(plugin): migrate tests to WordSpec

### PR DESCRIPTION
**Pull request by anarchuser**

### CODE REVIEW

File: plugin/src/test/sc/plugin2021/BoardTest.kt [hunk 1]
```diff
@@ -1,68 +1,86 @@
...
+    )
+    "Boards " When {
+        var one: Board
+        var two: Board
+        "constructed" Should {
+            one = Board()
+            two = Board()
+            "be separate instances" {
+                one shouldNotBeSameInstanceAs two
+            }
```
Comment (xerus2000) [inline plugin/src/test/sc/plugin2021/BoardTest.kt:24]: that's a weird check, but ok ^^

File: plugin/src/test/sc/plugin2021/BoardTest.kt [hunk 2]
```diff
@@ -1,68 +1,86 @@
...
+            Field(Coordinates(19, 2), FieldContent.YELLOW),
+            Field(Coordinates(8, 3), FieldContent.YELLOW),
+            Field(Coordinates(4, 9), FieldContent.YELLOW)
+    )
+    "Boards " When {
+        var one: Board
+        var two: Board
+        "constructed" Should {
+            one = Board()
+            two = Board()
```
Comment (xerus2000) [inline plugin/src/test/sc/plugin2021/BoardTest.kt:21]: when you use a proper `isolationMode` like I did in GameTest you can move that into common code

File: plugin/src/test/sc/plugin2021/BoardTest.kt [hunk 3]
```diff
@@ -1,68 +1,86 @@
...
+            "produce a list of differences" {
+                one.compare(two) shouldBe changingFields
+            }
+        }
+        "both are equally modified" Should {
+            one = Board()
+            two = Board()
+            changingFields.forEach { two[it.coordinates] = it.content }
+            changingFields.forEach { one[it.coordinates] = it.content }
+            "both be equal" {
```
Comment (xerus2000) [inline plugin/src/test/sc/plugin2021/BoardTest.kt:45]:
```suggestion
            "equal each other" {
```
to unify the terms

File: plugin/src/test/sc/plugin2021/ComparisonTest.kt [hunk 4]
```diff
@@ -2,25 +2,44 @@ package sc.plugin2021
...
+            position shouldNotBe Board()
+        }
     }
-    "Piece comparison" {
-        Rotation.values().map {
-            SetMove(Piece(kind = PieceShape.TETRO_O, rotation = it, isFlipped = true, position = Coordinates(7, 12)))
-            SetMove(Piece(kind = PieceShape.TETRO_O, rotation = it, isFlipped = false, position = Coordinates(7, 12)))
-        }.toSet().size shouldBe 1
+    "Piece comparison" should {
+        "look at coordinates only" {
```
Comment (xerus2000) [inline plugin/src/test/sc/plugin2021/ComparisonTest.kt:30]:
```suggestion
        "consider only the coordinates" {
```

File: plugin/src/test/sc/plugin2021/ComparisonTest.kt [hunk 5]
```diff
@@ -2,25 +2,44 @@ package sc.plugin2021
...
+            }.toSet() shouldHaveSize 1
+        }
     }
-    "SetMove comparison" {
-        Rotation.values().map {
-            SetMove(Piece(kind = PieceShape.TETRO_O, rotation = it, isFlipped = true, position = Coordinates(7, 12)))
-            SetMove(Piece(kind = PieceShape.TETRO_O, rotation = it, isFlipped = false, position = Coordinates(7, 12)))
-        }.toSet().size shouldBe 1
+    "SetMove comparison" should {
+        "look at their internal piece only" {
```
Comment (xerus2000) [inline plugin/src/test/sc/plugin2021/ComparisonTest.kt:38]:
```suggestion
        "consider only their internal piece" {
```

**Review by xerus2000 — APPROVED** [review]
That's okay, but I think we have more important things to do than converting the tests just because we can - I think it would be better to do it incrementally when we change something, especially considering that the plugin development will soon cease anyway.
If you do want to do this nonetheless, start with the common code, i.e. sdk and server.

**Comment by anarchuser** [comment]
> That's okay, but I think we have more important things to do than converting the tests just because we can - I think it would be better to do it incrementally when we change something, especially considering that the plugin development will soon cease anyway.
> If you do want to do this nonetheless, start with the common code, i.e. sdk and server.

I won't have much time to work on this, so I'd just like to go over some of my tests. Some of them are messy or may not be up to date anymore, and especially the plugin requires working tests

File: plugin/src/test/sc/plugin2021/BoardTest.kt (hunk 2 above)
Comment (anarchuser) [reply plugin/src/test/sc/plugin2021/BoardTest.kt:21]: I'm aware, this was more to get into the testing style. The whole "When ... constructed" is superfluous

**Review by xerus2000 — APPROVED** [review]
LGTM, tests succeed here as well

**Pull request merged**